tests/tracking/test_model_registry.py

```diff
@@ -0,0 +1,454 @@
+
+    # updating model version's stage changes last_updated_timestamp for registered model
+    rmd3 = mlflow_client.get_registered_model_details(name)
+    assert_is_between(start_time_0, end_time_0, rmd3.creation_timestamp)
+    assert_is_between(start_time_2, end_time_2, rmd3.last_updated_timestamp)
+
+    model_versions_detailed = [rm.latest_versions
+                               for rm in mlflow_client.list_registered_models() if rm.name == name]
+    assert 1 == len(model_versions_detailed)
+    assert set([1, 2]) == set([mvd.version for mvd in model_versions_detailed[0]])
+    assert set([name]) == set([mvd.registered_model.name for mvd in model_versions_detailed[0]])
+
+    # update description
+    start_time_3 = now()
+    mlflow_client.update_model_version(name=name, version=1, description="This is a test model")
+    end_time_3 = now()
+    mvd1c = mlflow_client.get_model_version_details(name, 1)
+    assert str(mvd1c.description) == "This is a test model"
+    assert_is_between(start_time_1, end_time_1, mvd1c.creation_timestamp)
+    assert_is_between(start_time_3, end_time_3, mvd1c.last_updated_timestamp)
+
+    # changing description for model version does not affect last_updated_timestamp for registered
+    # model
+    rmd4 = mlflow_client.get_registered_model_details(name)
+    assert_is_between(start_time_0, end_time_0, rmd4.creation_timestamp)
+    assert_is_between(start_time_2, end_time_2, rmd4.last_updated_timestamp)
+
+
+def test_delete_model_version_flow(mlflow_client, backend_store_uri):
+    name = 'DeleteMVTest'
+    start_time_0 = now()
+    mlflow_client.create_registered_model(name)
+    end_time_0 = now()
+    rmd1 = mlflow_client.get_registered_model_details(name)
+    assert_is_between(start_time_0, end_time_0, rmd1.creation_timestamp)
+    assert_is_between(start_time_0, end_time_0, rmd1.last_updated_timestamp)
+
+    start_time_1 = now()
+    mv1 = mlflow_client.create_model_version(name, "path/to/model", "run_id_1")
+    end_time_1 = now()
+    assert mv1.version == 1
+    assert mv1.registered_model.name == name
+    mvd1 = mlflow_client.get_model_version_details(name, 1)
+    assert_is_between(start_time_1, end_time_1, mvd1.creation_timestamp)
+    assert_is_between(start_time_1, end_time_1, mvd1.last_updated_timestamp)
+
+    # creating model version changes last_updated_timestamp for registered model
+    rmd2 = mlflow_client.get_registered_model_details(name)
+    assert_is_between(start_time_0, end_time_0, rmd2.creation_timestamp)
+    assert_is_between(start_time_1, end_time_1, rmd2.last_updated_timestamp)
+
+    mv2 = mlflow_client.create_model_version(name, "another_path/to/model", "run_id_1")
+    assert mv2.version == 2
+    assert mv2.registered_model.name == name
+    mv3 = mlflow_client.create_model_version(name, "a/b/c", "run_id_2")
+    assert mv3.version == 3
+    assert mv3.registered_model.name == name
+    model_versions_detailed = [rm.latest_versions
```

Comment:
do we have any tests explicitly calling `client.get_latest_versions` ?

Reply:
added